phoenix-core/src/main/java/org/apache/phoenix/transaction/OmidTransactionProvider.java

```diff
@@ -120,84 +101,20 @@ public void close() throws IOException {}
-        tsoConfig.setWaitStrategy(WAIT_STRATEGY.LOW_CPU.toString());
-
-        Injector injector = Guice.createInjector(new TSOMockModule(tsoConfig));
-        tso = injector.getInstance(TSOServer.class);
-        tso.startAsync();
-        tso.awaitRunning();
-
-        OmidClientConfiguration clientConfig = new OmidClientConfiguration();
-        clientConfig.setConnectionString("localhost:" + port);
-        clientConfig.setConflictAnalysisLevel(OmidClientConfiguration.ConflictDetectionLevel.ROW);
-
-        InMemoryCommitTable commitTable = (InMemoryCommitTable) injector.getInstance(CommitTable.class);
-
-        try {
-            // Create the associated Handler
-            TSOClient client = TSOClient.newInstance(clientConfig);
-
-            HBaseOmidClientConfiguration clientConf = new HBaseOmidClientConfiguration();
-            clientConf.setConnectionString("localhost:" + port);
-            clientConf.setConflictAnalysisLevel(OmidClientConfiguration.ConflictDetectionLevel.ROW);
-            clientConf.setHBaseConfiguration(config);
-            commitTableClient = commitTable.getClient();
-            commitTableWriter = commitTable.getWriter();
-            transactionManager = HBaseTransactionManager.builder(clientConf)
-                    .commitTableClient(commitTableClient)
-                    .commitTableWriter(commitTableWriter)
-                    .tsoClient(client).build();
-        } catch (IOException | InterruptedException e) {
-            throw new SQLExceptionInfo.Builder(
-                    SQLExceptionCode.TRANSACTION_FAILED)
-                    .setMessage(e.getMessage()).setRootCause(e).build()
-                    .buildException();
-        }
-
-        return new OmidTransactionService(tso, transactionManager);
-    }
-
-    static class OmidTransactionService implements PhoenixTransactionService {
-        private final HBaseTransactionManager transactionManager;
-        private TSOServer tso;
-
-        public OmidTransactionService(TSOServer tso, HBaseTransactionManager transactionManager) {
-            this.tso = tso;
-            this.transactionManager = transactionManager;
-        }
-
-        public void start() {
-
-        }
-
-        @Override
-        public void close() throws IOException {
-            if (transactionManager != null) {
-                transactionManager.close();
-            }
-            if (tso != null) {
-                tso.stopAsync();
-                tso.awaitTerminated();
-            }
-        }
+    // For testing only
+    public void injectTestService(HBaseTransactionManager transactionManager, CommitTable.Client commitTableClient) {
+        this.transactionManager = transactionManager;
+        this.commitTableClient = commitTableClient;
     }
 
     @Override
-    public Class<? extends RegionObserver> getCoprocessor() {
-        return OmidTransactionalProcessor.class;
```

Comment:
Any reason to use static string to specify the coprocessor class name?

Reply:
Originally I did this to handle missing classes better, but it's not really needed.
However, returning strings here simplifies the resto fo the code, as all HBase coprocessor admin methods expect and return class names.
I'm going to replace these with class.getName()